packages/grid/_modules_/grid/hooks/root/useEvents.ts

```diff
@@ -101,6 +102,8 @@ export function useEvents(gridRootRef: React.RefObject<HTMLDivElement>, apiRef:
   useGridApiOptionHandler(apiRef, GRID_COLUMN_HEADER_LEAVE, options.onColumnHeaderLeave);
   useGridApiOptionHandler(apiRef, GRID_COLUMN_ORDER_CHANGE, options.onColumnOrderChange);
 
+  useGridApiOptionHandler(apiRef, GRID_COLUMN_RESIZE, options.onColumnResize);
```

Comment:
same here

Reply:
here you can add `options.onColumnHeaderMouseDown`

Reply:
in the resize hook you catch `ColumnHeaderMouseDown` to make it a resize feature...
So the grid offers events and you can build anything with them

Reply:
I think that `options.onColumnResize` is better than `options.onColumnHeaderMouseDown`. Also there is a difference. I made it so that `options.onColumnResize` is fired while resizing.

Reply:
`options.onColumnResize` should stay but move to the resize hook.
My point was that if there was a mouseDown on column header then you could add 
`options.onColumnHeaderMouseDown` as well

Reply:
the goal of useEvents is to provide low level events to build feature on top. So it should be feature agnostic.
For example it should publish a cell click and not a cell selection as what you do on click is not down to useEvents, it is down useSelection, and it would be up to users to use useSelection plugin hook. Hope it's a bit clearer :)

Reply:
I still struggle to see the value for the `onColumnHeaderMouseDown`. Currently, there isn't a `mouseDown` on a column header, there is one on the separator only. The column header has a click only.

Regarding moving the handler - will do.

Reply:
Following the example from here https://material-ui.com/api/slider/ we can do an `onColumnResize` (as it is at the moment) and add `onColumnResizeCommitted` that is called only once when the resizing stops (on `mouseup`)?

Reply:
Updated